### DISCUSSION TITLE
Pagination for 1M rows

### DISCUSSION THREAD

**Discussion by shukladheeraj01** [jxlsteam/jxls#160] [Q&A] [answered]
Hello, 

I have a requirement to write 1M records to excel with multi-sheet  template formatting, Is there any way to achieve this in paginated way? Since writing 1M records in one go may turn into OOM Exception.
Is there any way to achieve it in optimal way?

**Comment by SoltauFintel** [comment, marked as answer]
You can write 1 million rows in 1 sheet by using Streaming.

Writing 1 million rows - even in several sheets - without Streaming will result in an OOM error.

https://github.com/jxlsteam/jxls/blob/a2c69bedfdbf01067e9e1d6d4df598eff8b71c8c/jxls-poi/src/test/java/org/jxls/templatebasedtests/TableTest.java
